OfficeIMO.Examples/Word/Fields/Fields.cs

```diff
@@ -0,0 +1,42 @@
+using System;
+using System.Collections.Generic;
+using DocumentFormat.OpenXml.Drawing;
+using DocumentFormat.OpenXml.Wordprocessing;
+using OfficeIMO.Word;
+
+namespace OfficeIMO.Examples.Word {
+    internal static partial class Fields {
+        internal static void Example_DocumentWithFields(string folderPath, bool openWord) {
+            Console.WriteLine("[*] Opening Document with fields");
+            var filePath = System.IO.Path.Combine(System.IO.Directory.GetCurrentDirectory(), "Templates", "DocumentWithFields.docx");
+            
+
+            using (WordDocument document = WordDocument.Load(filePath)) {
+
+                var firstField = document.Fields[0];
+                Console.WriteLine("...Type: " + firstField.FieldType);
+                Console.WriteLine("...Format switch: " + firstField.FieldFormat);
+                Console.WriteLine("...Switches: " + String.Join(" ",firstField.FieldSwitches));
+
+                var tocPageRef = document.Fields[1];
+                Console.WriteLine("...Type: " + tocPageRef.FieldType);
+                Console.WriteLine("...Format switch: " + tocPageRef.FieldFormat);
+                Console.WriteLine("...Instruction: " + String.Join(" ", tocPageRef.FieldInstructions));
+                Console.WriteLine("...Switches: " + String.Join(" ", tocPageRef.FieldSwitches));
+
+                var askField = document.Fields.Find((e)=>e.FieldType == WordFieldType.Ask);
+                Console.WriteLine("...Type: " + askField.FieldType);
+                Console.WriteLine("...Format switch: " + askField.FieldFormat);
+                Console.WriteLine("...Switches: " + String.Join(" ", askField.FieldSwitches));
+
+                //Replace ask field with new question
+                askField.Remove();
+                var bookmark = document.Bookmarks.ToArray()[0];
+                document.AddField(WordFieldType.Ask, parameters: new List<String> { bookmark.Name.ToString(), "\"How was your day?\"", "\\d \"Thanks for asking\"" });
```

Comment:
I wonder if this could be further 'simplified'? I would like to avoid users making mistakes by having to type special chars and knowing what those special chars mean for any given type of a field. While on initial implementation for "hardcore" guys like you this is probably ok, do you think we could create simplified versions of it such as document.AddFieldAsk, document.AddFieldBib.. with parameters that make sense for mortals without need to know OpenXML specification? 

Maybe I am just weak programmer but my idea for OfficeIMO is that I would like to avoid knowing openxml, like at all to use it. Like I mentioned before I mostly did OfficeIMO for PSwriteOffice as a PowerShell module and PowerShell is mostly used by non-programmers. 

What do you think?

Reply:
So the idea would be - we get this in as is, and open an issue with your whole idea and simply once you have some spare time you create PR after PR. Whether it takes weeks, months or years - the goal is defined ;-) 

The whole OfficeIMO project is spare time for me. Most of my projects are 50% spare time, 50% work related.